File: 383-ransom-note/383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
          int a[]=new int[26];
          //cout<<a[magazine[0]-'a'];
        for(int i=0;i<magazine.length();++i)
            a[magazine.charAt(i)-'a']++;
        for(int i=0;i<ransomNote.length();++i)
             if(--a[ransomNote.charAt(i)-'a']<0)
                 return false;
        return true;
    }
}

File: 383-ransom-note/383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int> a(26,0);
          //cout<<a[magazine[0]-'a'];
        for(int i=0;i<magazine.size();++i)
            a[magazine[i]-'a']++;
        for(int i=0;i<ransomNote.size();++i)
             if(--a[ransomNote[i]-'a']<0)
                 return false;
        return true;
      
    }
};